app/src/screens/PersonCredentialLoading.tsx

```diff
@@ -0,0 +1,258 @@
+  ScrollView,
+  StyleSheet,
+  Text,
+  View,
+} from 'react-native'
+
+import PersonCredentialSpinner from '../components/PersonCredentialSpinner'
+import { connectToIASAgent, authenticateWithServiceCard, WellKnownAgentDetails } from '../helpers/BCIDHelper'
+import { useAttestation } from '../hooks/useAttestation'
+import { AttestationEventTypes } from '../services/attestation'
+import { BCState } from '../store'
+
+export default function PersonCredentialLoading() {
+  const { ColorPallet, TextTheme } = useTheme()
+  const [store] = useStore<BCState>()
+  const [remoteAgentDetails, setRemoteAgentDetails] = useState<WellKnownAgentDetails | undefined>()
+  const receivedProofRequests = useProofByState(ProofState.RequestReceived)
+  const timer = useRef<NodeJS.Timeout>()
+  const logger = useContainer().resolve(TOKENS.UTIL_LOGGER)
+  const receivedCredentialOffers = useCredentialByState(CredentialState.OfferReceived)
+  const { loading: attestationLoading } = useAttestation()
+  const { agent } = useAgent()
+  if (!agent) {
+    throw new Error('Unable to fetch agent from Credo')
+  }
+  const navigation = useNavigation()
+  const { t } = useTranslation()
+  const attestationProofRequestWaitTimeout = 10000
+  const [didCompleteAttestationProofRequest, setDidCompleteAttestationProofRequest] = useState<boolean>(false)
+
+  const styles = StyleSheet.create({
+    container: {
+      height: '100%',
+      backgroundColor: ColorPallet.brand.modalPrimaryBackground,
+      padding: 20,
+    },
+    image: {
+      marginTop: 80,
+    },
+    messageContainer: {
+      alignItems: 'center',
+      marginTop: 40,
+    },
+    messageText: {
+      fontWeight: TextTheme.normal.fontWeight,
+      textAlign: 'center',
+      marginTop: 30,
+    },
+    controlsContainer: {
+      marginTop: 'auto',
+      margin: 20,
+    },
+    delayMessageText: {
+      textAlign: 'center',
+      marginTop: 20,
+    },
+  })
+
+  useEffect(() => {
+    const connect = async () => {
+      try {
+        const remoteAgentDetails = await connectToIASAgent(agent, store, t)
+        setRemoteAgentDetails(remoteAgentDetails)
+        logger.error(`Connected to IAS agent, connectionId: ${remoteAgentDetails.connectionId}`)
+      } catch (err) {
+        logger.error(`Failed to connect to IAS agent, error: ${(err as BifoldError).message}`)
+      }
+    }
+    connect()
+  }, [])
+
+  useEffect(() => {
+    if (!remoteAgentDetails) {
+      return
+    }
+
+    timer.current = setTimeout(() => {
+      const proofRequest = receivedProofRequests.find((proof) => proof.connectionId === remoteAgentDetails.connectionId)
+
+      if (!proofRequest) {
+        // No proof from our IAS Agent to respond to, do nothing.
+        logger.info(`Waited ${attestationProofRequestWaitTimeout / 1000}sec on attestation proof request, continuing`)
+
+        setDidCompleteAttestationProofRequest(true)
+      }
+    }, attestationProofRequestWaitTimeout)
+  }, [remoteAgentDetails])
```

Comment:
```suggestion
  }, [remoteAgentDetails, receivedProofRequests, attestationProofRequestWaitTimeout, logger]);
```

Reply:
attestationProofRequestWaitTimeout and logger should be constant for the duration of the screen no? I don't think its possible for either of them to change without reloading the app